mu-plugins/central-hub/tests/phpunit/integration/meta-data/saveMetaBoxes.php

```diff
@@ -0,0 +1,374 @@
+	}
+
+	/**
+	 * Test save_meta_boxes() should not save when custom fields config (config store) and
+	 *    post meta values in $_POST are empty.
+	 */
+	public function test_should_not_save_when_custom_fields_config_and_post_meta_in_POST_are_empty() {
+		// By default, $config is an empty array when the config store is empty.
+		// Add post meta to the database.
+		add_post_meta( $this->post, 'role', 'Bass/Baritone' );
+		add_post_meta( $this->post, 'residence_city', 'Houston' );
+		add_post_meta( $this->post, 'residence_state', 'TX' );
+		add_post_meta( $this->post, 'tour_number', '1' );
+		$_POST = [
+			'post_ID'            => $this->post,
+			'post_status'        => 'publish',
+			'members'            => [
+				'role'            => '',
+				'residence_city'  => '',
+				'residence_state' => '',
+				'tour_number'     => '',
+			],
+			'members_nonce_name' => wp_create_nonce( 'members_nonce_action' ),
+		];
+		$this->assertSame( 'Bass/Baritone', get_post_meta( $this->post, 'role', true ) );
+		$this->assertSame( 'Houston', get_post_meta( $this->post, 'residence_city', true ) );
+		$this->assertSame( 'TX', get_post_meta( $this->post, 'residence_state', true ) );
+		$this->assertSame( '1', get_post_meta( $this->post, 'tour_number', true ) );
+
+		save_meta_boxes( $this->post );
+
+		$this->assertSame( 'Bass/Baritone', get_post_meta( $this->post, 'role', true ) );
+		$this->assertSame( 'Houston', get_post_meta( $this->post, 'residence_city', true ) );
+		$this->assertSame( 'TX', get_post_meta( $this->post, 'residence_state', true ) );
+		$this->assertSame( '1', get_post_meta( $this->post, 'tour_number', true ) );
+	}
+
+	/**
+	 * Test save_meta_boxes() should not save when the store_key does not start with `meta_box.`.
+	 */
+	public function test_should_not_save_when_store_key_does_not_start_with_meta_box() {
+		$configs = [
+			'notametabox.members' => [
+				'custom_fields' => [
+					'role'            => 'Soprano',
+					'residence_city'  => 'Chicago',
+					'residence_state' => 'IL',
+					'tour_number'     => '3',
+				],
+			],
+		];
+		foreach ( $configs as $store_key => $config_to_store ) {
+			loadConfig( $store_key, $config_to_store );
+		}
+		$_POST = [
+			'post_ID'            => $this->post,
+			'post_status'        => 'publish',
+			'members'            => [
+				'role'            => '',
+				'residence_city'  => '',
+				'residence_state' => '',
+				'tour_number'     => '',
+			],
+			'members_nonce_name' => wp_create_nonce( 'members_nonce_action' ),
+		];
+		// Before function call, check that database is empty of post meta for this post.
+		$this->assertSame( '', get_post_meta( $this->post, 'role', true ) );
+		$this->assertSame( '', get_post_meta( $this->post, 'residence_city', true ) );
+		$this->assertSame( '', get_post_meta( $this->post, 'residence_state', true ) );
+		$this->assertSame( '', get_post_meta( $this->post, 'tour_number', true ) );
+
+		$this->assertNull( save_meta_boxes( $this->post ) );
```

Comment:
@rgadon107 I’d suggest removing the assertion here and just invoking the save function.

Reply:
Glad you provided a second set of eyes to look the file over. I missed that 2nd assertion. Done.